test/unit/system/prepare_test.py

```diff
@@ -304,6 +304,86 @@ def test_install_bootstrap(
         )
 
+    @patch('kiwi.system.prepare.RootInit')
+    @patch('kiwi.system.prepare.RootBind')
+    @patch('kiwi.logger.Logger.get_logfile')
+    @patch('kiwi.xml_state.XMLState.get_bootstrap_collection_type')
+    @patch('kiwi.system.prepare.CommandProcess.poll_show_progress')
+    @patch('kiwi.system.prepare.ArchiveTar')
+    @patch('os.path.exists')
+    def test_install_bootstrap_archive_target_dir(
+        self, mock_exists, mock_tar, mock_poll, mock_collection_type,
+        mock_get_logfile, mock_root_bind, mock_root_init
+    ):
+        Defaults.set_platform_name('x86_64')
+        mock_get_logfile.return_value = None
+        description = XMLDescription(
+            description='../data/example_config_target_dir.xml',
+            derived_from='derived/description'
+        )
+        self.description_dir = os.path.dirname(description.description_origin)
+        self.xml = description.load()
+
+        self.manager = MagicMock(
+            return_value=MagicMock()
+        )
+        self.manager.package_requests = ['foo']
+        self.manager.collection_requests = ['foo']
+        self.manager.product_requests = ['foo']
+
+        root_init = MagicMock()
+        mock_root_init.return_value = root_init
+        root_bind = MagicMock()
+        root_bind.root_dir = 'root_dir'
+        mock_root_bind.return_value = root_bind
+        self.state = XMLState(
+            self.xml
+        )
+        self.system = SystemPrepare(
+            xml_state=self.state, root_dir='root_dir',
+            allow_existing=True
+        )
+        mock_root_init.assert_called_once_with(
+            'root_dir', True
+        )
+        root_init.create.assert_called_once_with()
+        mock_root_bind.assert_called_once_with(
+            root_init
+        )
+        root_bind.setup_intermediate_config.assert_called_once_with()
+        root_bind.mount_kernel_file_systems.assert_called_once_with()
+        # set the xml for archive target done
+
+        mock_exists.return_value = True
+        tar = Mock()
+        tar.extract = Mock()
+        mock_tar.return_value = tar
+        mock_collection_type.return_value = 'onlyRequired'
+        self.system.install_bootstrap(self.manager)
+        self.manager.process_only_required.assert_called_once_with()
+        self.manager.request_package.assert_any_call(
+            'filesystem'
+        )
+        self.manager.request_package.assert_any_call(
+            'zypper'
+        )
+        self.manager.request_collection.assert_called_once_with(
+            'bootstrap-collection'
+        )
+        self.manager.request_product.assert_called_once_with(
+            'kiwi'
+        )
+        self.manager.process_install_requests_bootstrap.assert_called_once_with(
+            self.system.root_bind
+        )
+        mock_tar.assert_called_once_with(
+            '{0}/bootstrap.tgz'.format(self.description_dir)
+        )
+        tar.extract.assert_called_once_with('root_dir/foo')
```

Comment:
This is a 99% copy of the test_install_bootstrap() test. I think it's not needed to have all the other assertions tested again in this test. I suggest to reduce the test to just the relevant line testing for tar.extract and how it is called.

I understand that the copy&paste was the fastest way to make this test but it will strike back if code changes causes adaptions of tests. So please let us avoid the duplication. Makes sense ?